tests/compiler/function-expression.optimized.wat

```diff
@@ -1,5 +1,99 @@
 (module
+ (type $i32_i32_=>_i32 (func (param i32 i32) (result i32)))
+ (type $none_=>_none (func))
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (type $i32_i32_i32_i32_=>_none (func (param i32 i32 i32 i32)))
+ (type $none_=>_i32 (func (result i32)))
+ (type $i32_i32_i32_=>_i32 (func (param i32 i32 i32) (result i32)))
+ (import "env" "abort" (func $~lib/builtins/abort (param i32 i32 i32 i32)))
  (memory $0 1)
  (data (i32.const 1024) ",\00\00\00\01\00\00\00\01\00\00\00,\00\00\00f\00u\00n\00c\00t\00i\00o\00n\00-\00e\00x\00p\00r\00e\00s\00s\00i\00o\00n\00.\00t\00s")
+ (table $0 12 funcref)
+ (elem (i32.const 1) $start:function-expression~anonymous|0~nonClosure $start:function-expression~anonymous|0~nonClosure $start:function-expression~someName~nonClosure $start:function-expression~anonymous|2~nonClosure $start:function-expression~anonymous|3~nonClosure $start:function-expression~anonymous|4~nonClosure $start:function-expression~anonymous|5~nonClosure $start:function-expression~anonymous|3~nonClosure $start:function-expression~anonymous|4~nonClosure $start:function-expression~anonymous|5~nonClosure $start:function-expression~anonymous|2~nonClosure)
  (export "memory" (memory $0))
+ (start $~start)
+ (func $start:function-expression~anonymous|0~nonClosure (param $0 i32) (result i32)
+  local.get $0
+ )
+ (func $start:function-expression~someName~nonClosure
+  nop
+ )
+ (func $start:function-expression~anonymous|2~nonClosure (result i32)
+  i32.const 1
+ )
+ (func $start:function-expression~anonymous|3~nonClosure (param $0 i32) (param $1 i32) (result i32)
+  local.get $0
+  local.get $1
+  i32.add
+ )
```

Comment:
Need to look into why this doesn't get optimized away anymore 🤔

Reply:
Typically `call`ed somewhere directly, or becomes inserted into the table and there is at least one `call_indirect` expression somewhere that Binaryen cannot precompute. Exporting the table also prevents it, but I think that's not what's happening here.

Reply:
Ah right.. I suppose there's no way for binaryen to precompute `call_indirect` if it's calling a function pointer stored in linear memory